daffodil-core/src/main/scala/org/apache/daffodil/core/compiler/Compiler.scala

```diff
@@ -110,7 +110,7 @@ final class ProcessorFactory private (
   override def onPath(xpath: String): DFDL.DataProcessor = sset.onPath(xpath)
 
   override def forLanguage(language: String): DFDL.CodeGenerator = {
-    Assert.usage(!isError)
+    Assert.usage(!isError, getDiagnostics.mkString)
```

Comment:
`Assert.usage` means that the user calling this API used he API incorrectly. So instead of displaying the diagnostics associated with the failed ProcessorFactory, I think the error message wants to be about incorrect API usage. In this case a usage failure means the user did not check `isError` before calling the `forLanguage` function. So the message probably wants to instead be something like:

> ProcessorFactory forLanguage() must not be called when isError is true.

I'm not sure I love that, but something along those lines.